File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

//Approach-1 (Using O(n) space)
//T.C : O(n)
//S.C : O(n)
var inorderTraversal = function(root) {
   let result=[];
    
    // Helper function to traverse the tree in inorder
    const inorder = (node) => {
        if (node === null) {
            return;
        }
        inorder(node.left); // Traverse left subtree
        result.push(node.val); // Add the value to the result
        inorder(node.right); // Traverse right subtree
    };
    
  inorder(root); // Perform in-order traversal
return   result;
};

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>() ;
        inOrder(root, result);
        return result;        
    }

    public void inOrder(TreeNode curr,  List<Integer> result){
        if(curr == null) return;
        inOrder(curr.left,result );
        result.add(curr.val);
        inOrder(curr.right,result );
    }
}